### ISSUE TITLE
Change constructor signature of MultiChannelPulseTemplate and SequencePulseTemplate

### ISSUE THREAD

**Issue by terrorfisch** [qutech/qc-toolkit#138]
Make external_parameters an optional keyword only argument and all subtemplates positional

Current:
```
def __init__(self,
             subtemplates: Iterable[Union[Tuple, PulseTemplate]],
             external_parameters: Union[Iterable[str], Set[str]],
             identifier: Optional[str]=None)

SequencePulseTemplate([pt_1, pt_2, (pt_1, {'a': 'b*c'})],
                      {'a', 'b', 'c', 'pt_2_parameter'})
SequencePulseTemplate([pt_1, pt_2, (pt_1, {'a': 'b*c'})],
                     external_parameters={'a', 'b', 'c', 'pt_2_parameter'})

#Error since no external parameters
SequencePulseTemplate([pt_1, pt_2, (pt_1, {'a': 'b*c'})])
```

Proposed:
```
def __init__(self,
             *subtemplates: Tuple[Union[Tuple, PulseTemplate]],
             external_parameters: Optional[Union[Iterable[str], Set[str]]]=None,
             identifier: Optional[str]=None)

SequencePulseTemplate(pt_1, pt_2, (pt_1, {'a': 'b*c'}),
                      external_parameters={'a', 'b', 'c', 'pt_2_parameter'})
SequencePulseTemplate(pt_1, pt_2, (pt_1, {'a': 'b*c'}))

#Error since external paramters is keyword only
SequencePulseTemplate(pt_1, pt_2, (pt_1, {'a': 'b*c'}),
                      {'a', 'b', 'c', 'pt_2_parameter'})
```

Pros:
1. Easier to type
2. Last chance for interface changes

Cons:
1. Interface change
2. Consitency checks not forced
3. `__matmul__` is without consitency checks if needed

**Comment by terrorfisch** [comment]
Also see #74

**Comment by terrorfisch** [comment]
Implemented in #143 except ```__matmul__```
